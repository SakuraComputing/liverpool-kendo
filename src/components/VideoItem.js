import React from 'react';
import ReactPlayer from 'react-player';

class VideoItem extends React.Component {

    render() {

        const { title, youTubeUrl, synopsis} = this.props;

        return (

            <div className="content-container">
                <h2>{title}</h2>
                <div className="player-wrapper">
                    <ReactPlayer
                        className='react-player'
                        url={youTubeUrl}
                        width='100%'
                        height='100%'
                        controls={true}
                    />
                </div>
                <div>{synopsis}</div>
            </div>
        );
    }

};

export default VideoItem;
