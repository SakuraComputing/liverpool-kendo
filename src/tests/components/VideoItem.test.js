import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import VideoItem from '../../components/VideoItem';

test('should render the Video page',() => {
    const props = {
      title: 'Japanology',
      youTubeUrl: 'https://www.youtube.com/watch?v=8Ea2ih7uNW4',
      synopsis: '30 minute program'
    };
    const wrapper = shallow(<VideoItem props={props}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
