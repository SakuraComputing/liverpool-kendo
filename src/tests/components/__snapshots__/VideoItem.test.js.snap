// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Video page 1`] = `
<div
  className="content-container"
>
  <h2 />
  <div
    className="player-wrapper"
  >
    <ReactPlayer
      className="react-player"
      config={
        Object {
          "dailymotion": Object {
            "params": Object {
              "api": 1,
              "endscreen-enable": false,
            },
            "preload": false,
          },
          "facebook": Object {
            "appId": "1309697205772819",
          },
          "file": Object {
            "attributes": Object {},
            "forceAudio": false,
            "forceDASH": false,
            "forceHLS": false,
            "hlsOptions": Object {},
            "tracks": Array [],
          },
          "soundcloud": Object {
            "options": Object {
              "buying": false,
              "download": false,
              "liking": false,
              "sharing": false,
              "show_comments": false,
              "show_playcount": false,
              "visual": true,
            },
          },
          "vimeo": Object {
            "playerOptions": Object {
              "autopause": false,
              "autoplay": false,
              "byline": false,
              "portrait": false,
              "title": false,
            },
            "preload": false,
          },
          "wistia": Object {
            "options": Object {},
          },
          "youtube": Object {
            "playerVars": Object {
              "autoplay": 0,
              "iv_load_policy": 3,
              "modestbranding": 1,
              "playsinline": 1,
              "rel": 0,
              "showinfo": 0,
            },
            "preload": false,
          },
        }
      }
      controls={true}
      height="100%"
      loop={false}
      muted={false}
      onBuffer={[Function]}
      onDuration={[Function]}
      onEnded={[Function]}
      onError={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onProgress={[Function]}
      onReady={[Function]}
      onSeek={[Function]}
      onStart={[Function]}
      playbackRate={1}
      playing={false}
      playsinline={false}
      progressInterval={1000}
      style={Object {}}
      volume={0.8}
      width="100%"
    />
  </div>
  <div />
</div>
`;
